src/runtime/containerd-shim-v2/stream.go

```diff
@@ -109,7 +109,6 @@ func ioCopy(exitch, stdinCloser chan struct{}, tty *ttyIO, stdinPipe io.WriteClo
 			defer bufPool.Put(p)
 			io.CopyBuffer(tty.Stdout, stdoutPipe, *p)
 			wg.Done()
-			closeOnce.Do(tty.close)
```

Comment:
You can't  delete this tty close, since it will close the tty's stdin and then the following wg.Wait() can continue, otherwise, the stdin iocopy wouldn't ended and there's a dead lock occur.  The root issue here is that it shouldn't close the tty's stderr at the end of the stdout iocopy, so in addition to delete "closeOnce.Do(tty.close)",  you should also close the tty's stdin here.

Reply:
Right. In my tests, stdin was nil so the routine was not started, and I couldn't see this deadlock.
I will change that. Thanks!